main.py

```diff
@@ -759,20 +763,50 @@ def handle_link_up(self, event):
 
     # Possibly replace this with interruptions?
     @listen_to(
-        '.*.switch.interface.(link_up|link_down|created|deleted)',
-        pool='dynamic_single'
+        '.*.switch.interface.(link_up|link_down|created|deleted)'
     )
     def on_interface_link_change(self, event: KytosEvent):
         """
-        Handler for interface link_up and link_down events
+        Handler for interface link_up and link_down events.
+
+        To avoid multiple database updated (link flap):
+        First check the time when interface was last updated.
+        Second check the time when the last interface event was received.
+        Last event received always updates database.
         """
-        with self._lock:
-            _, _, event_type = event.name.rpartition('.')
-            iface = event.content.get("interface")
-            if event_type in ('link_up', 'created'):
-                self.handle_interface_link_up(iface)
-            elif event_type in ('link_down', 'deleted'):
-                self.handle_interface_link_down(iface)
+        iface = event.content.get("interface")
+        self.intf_events[iface.id].update({"received": now()})
+        with self.lock_interfaces[iface.id]:
+            last_updated = self.intf_events[iface.id].get("updated")
+            _now = now()
+            if not last_updated:
+                self.intf_events[iface.id].update({"updated": _now})
+                self.handle_on_interface_link_change(event)
+            elif int((_now - last_updated).total_seconds() * 1000) >= 100:
+                last_updated = last_updated.replace(tzinfo=timezone.utc)
+                self.intf_events[iface.id].update({"updated": _now})
+                self.handle_on_interface_link_change(event)
+
+        time.sleep(0.1)
+        with self.lock_interfaces[iface.id]:
+            last_received = self.intf_events[iface.id].get("received")
+            _now = now()
+            last_received = last_received.replace(tzinfo=timezone.utc)
+            if int((_now - last_received).total_seconds() * 1000) < 100:
```

Comment:
I think there's a corner case here where a potential out of order KytosEvent being processed here might end up being discarded, which could leave the interface in an inconsistent state. Also multipling by `1000` here doesn't look coherent. Can you verify if this is left offer from the first iteration you were using? 

Generally speaking, I believe the safest (implementation) route would be:

- First, solve all the potential out of the order events [like we do in `topology`](https://github.com/kytos-ng/topology/blob/master/main.py#L960-L965), always relying on `event.timestamp` as a reference since we don't have a (deterministic) guarantee when the decorated threaded handler will actually execute due to thread pool scheduling/preemption, so relying on a `now()` from a handler might not completely give the reference point you needed in certain cases. 
- Second, keep appending the events in memory on a deque, and always process the most recent one (which we'll have a guarantee that's been ordered), then you can only pop every X (0.1) seconds, and after processing it, then you keep popping left any older event than the one that's just been processed and keep repeating this process. 

Essentially this would keep processing as it it were a sliding time-based window. Let me know if you can see another simpler and safer implementation than this one here that I mentioned.

Reply:
Returning out of order events.
Now only applying last event per interface every 0.1 seconds.